src/client.rs

```diff
@@ -84,6 +84,84 @@ pub struct Connection<T: Read + Write> {
     pub greeting_read: bool,
 }
 
+/// A builder for the append command
+pub struct AppendCmd<'a, T: Read + Write> {
+    session: &'a mut Session<T>,
+    content: &'a [u8],
+    mailbox: &'a str,
+    flags: Vec<Flag<'a>>,
+    date: Option<DateTime<FixedOffset>>,
+}
+
+impl<'a, T: Read + Write> AppendCmd<'a, T> {
+    /// The [`APPEND` command](https://tools.ietf.org/html/rfc3501#section-6.3.11) can take
+    /// an optional FLAGS parameter to set the flags on the new message.
+    ///
+    /// > If a flag parenthesized list is specified, the flags SHOULD be set
+    /// > in the resulting message; otherwise, the flag list of the
+    /// > resulting message is set to empty by default.  In either case, the
+    /// > Recent flag is also set.
+    ///
+    /// The [`\Recent` flag](https://tools.ietf.org/html/rfc3501#section-2.3.2) is not
+    /// allowed as an argument to `APPEND` and will be filtered out if present in `flags`.
+    pub fn flag(&mut self, flag: Flag<'a>) -> &mut Self {
+        self.flags.push(flag);
+        self
+    }
+
+    /// Append an array of flags
+    pub fn flags(&mut self, flags: &'a [Flag<'a>]) -> &mut Self {
+        self.flags.append(&mut flags.to_vec());
+        self
+    }
+
+    /// Pass a date in order to set the date that the message was originally sent.
+    ///
+    /// > If a date-time is specified, the internal date SHOULD be set in
+    /// > the resulting message; otherwise, the internal date of the
+    /// > resulting message is set to the current date and time by default.
+    pub fn internal_date(&mut self, date: DateTime<FixedOffset>) -> &mut Self {
+        self.date = Some(date);
+        self
+    }
+
+    /// Run command
+    #[must_use = "always run a command once options are configured"]
+    pub fn run(&mut self) -> Result<()> {
```

Comment:
I wonder if this function should instead be called `finish`?

Reply:
Sounds good to me.